src/globus_sdk/exc/warnings.py

```diff
@@ -0,0 +1,45 @@
+import os
+import typing as t
+import warnings
+
+
+# this is (by meaning) a type of DeprecationWarning, but because python suppresses
+# DeprecationWarning by default, most users would never see it if we inherited from that
+# therefore, inheriting from Warning is an "incorrect" but pragmatic choice
+#
+# in concert with the helpers below to make the warnings opt-in, this results in good
+# user-facing behavior
+class GlobusSDKLegacyBehaviorWarning(Warning):
+    def __init__(
+        self,
+        message: str,
+        *args: t.Any,
+        will_be_removed: str | None = "4.0.0",
```

Comment:
The behaviors around this are somewhat hard to access via warning emission, so I also considered making a multi-class hierarchy here, like
```python
class Foo(Warning):
    will_be_removed: str | None = None

class Bar(Foo):
    will_be_removed = "4.0.0"
```

but it seemed like overthinking things. I think we should feel okay about rewriting this class a bit if we need to in the future.

Reply:
```suggestion
        removal_version: str | None = "4.0.0",
```

Not sure if my suggestion is perfect but I don't think `will_be_removed` conveys what this thing is well enough.

Reply:
I think "removal_version" may be a better name, but after our discussion and doing a refactor, I'm even happier with where this ended up (with a simple empty warning class whose name encodes this notion).